forgevtt-module.js

```diff
@@ -1989,23 +2012,48 @@ class ForgeVTT_FilePicker extends FilePicker {
         return buckets;
     }
 
+    /**
+     * Retrieves the Forge VTT bucket based on the provided key or index.
+     * @param {string|number} bucketKey - The key or index of the bucket to retrieve.
+     * @returns {object} The Forge VTT bucket object.
+     */
+    static _getForgeVttBucket(bucketKey) {
+        this._forgeBucketIndex = this._forgeBucketIndex || this._getForgeVTTBuckets();
+        // From Foundry v12, buckets are keyed by index not by hash
+        const isHashKey = isNaN(bucketKey) || !foundry.utils.isNewerVersion(ForgeVTT.foundryVersion, "12");
+        const bucketIndex = isHashKey ? this._forgeBucketIndex.findIndex((b) => b.key === bucketKey) : bucketKey;
+        return this._forgeBucketIndex[bucketIndex];
+    }
```

Comment:
Because the select value, and the `bucket` references in turn, are numbered keys in v12 rather than hashKeys, this function is used to get a bucket for either key formats.